pkg/provider/azure_storageaccount.go

```diff
@@ -118,25 +118,54 @@ func (az *Cloud) GetStorageAccesskey(account, resourceGroup string) (string, err
 	return "", fmt.Errorf("no valid keys")
 }
 
+// CheckAndCreateStorageAccount validates if the storage account exists, if not, creates a new storage account.
+// Returns storageAccountKey.
+func (az *Cloud) CheckAndCreateStorageAccount(accountOptions *AccountOptions) (string, error) {
+	if accountOptions == nil {
+		return "", fmt.Errorf("account options is nil")
+	}
+
+	accountName := accountOptions.Name
+
+	if len(accountName) == 0 {
+		return "", fmt.Errorf("Account name not provided in account options")
+	} else {
+		accountExists, err := az.checkIfAccountExists(accountOptions)
```

Comment:
if you only need `checkIfAccountNotExists`, I think you could add this function into your CSI driver, it's not necessary to add this function here.

Reply:
I'll move the function `checkIfAccountNotExists` to the driver. But I would need the function to create the storage account.

Reply:
what about add a new field "CreateIfNotExists"(by default: false) in AccountOptions, that would be a minor change

Reply:
The EnsureStorageAccount function is designed in a way to work when accountName is empty in accountOptions. For my case, I would need to create an account with that particular accountName if it doesnt exist. So, the `createStorageAccount` function might be needed after all?